File: leetcode/240-search-2d-matrix.cpp
#include <vector>

using namespace std;

class Solution {
public:
  bool searchMatrix(const vector<vector<int>> &matrix, int target) {
    const int height = matrix.size(), width = matrix.at(0).size();
    int i = 0, j = width - 1;
    while (i < height && j >= 0) {
      int val = matrix[i][j];
      if (val == target)
        return true;
      if (val < target)
        i += 1;
      if (val > target)
        j -= 1;
    }
    return false;
  }
};

File: leetcode/240-search-2d-matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        i = 0
        j = len(matrix[0]) - 1
        while True:
            try:
                n = matrix[i][j]
            except IndexError:
                return False
            if n == target:
                return True
            if n > target:
                j -= 1
            if n < target:
                i += 1
        return False
